samples/java/hibernate/src/main/java/com/google/cloud/postgres/HibernateSampleTest.java

```diff
@@ -0,0 +1,294 @@
+      CriteriaDelete<Albums> albumsCriteriaDelete = cb.createCriteriaDelete(Albums.class);
+      Root<Albums> albumsRoot = albumsCriteriaDelete.from(Albums.class);
+      albumsCriteriaDelete.where(cb.equal(albumsRoot.get("id"), albums.getId()));
+      Transaction transaction = s.beginTransaction();
+      s.createQuery(albumsCriteriaDelete).executeUpdate();
+      transaction.commit();
+    }
+  }
+
+  public void testJPACriteria() {
+    try (Session s = hibernateConfiguration.openSession()) {
+      CriteriaBuilder cb = s.getCriteriaBuilder();
+      CriteriaQuery<Singers> singersCriteriaQuery = cb.createQuery(Singers.class);
+      Root<Singers> singersRoot = singersCriteriaQuery.from(Singers.class);
+      singersCriteriaQuery
+          .select(singersRoot)
+          .where(
+              cb.and(
+                  cb.equal(singersRoot.get("firstName"), "David"),
+                  cb.equal(singersRoot.get("lastName"), "Lee")));
+
+      Query<Singers> singersQuery = s.createQuery(singersCriteriaQuery);
+      List<Singers> singers = singersQuery.getResultList();
+
+      System.out.println("Listed singer: " + singers.size());
+
+      CriteriaUpdate<Albums> albumsCriteriaUpdate = cb.createCriteriaUpdate(Albums.class);
+      Root<Albums> albumsRoot = albumsCriteriaUpdate.from(Albums.class);
+      albumsCriteriaUpdate.set("marketingBudget", new BigDecimal("5.0"));
+      albumsCriteriaUpdate.where(cb.equal(albumsRoot.get("id"), UUID.fromString(albumsId.get(0))));
+      Transaction transaction = s.beginTransaction();
+      s.createQuery(albumsCriteriaUpdate).executeUpdate();
+      transaction.commit();
+    }
+  }
+
+  public void testHqlUpdate() {
+    try (Session s = hibernateConfiguration.openSession()) {
+      Singers singers = Utils.createSingers();
+      singers.setLastName("Cord");
+      s.getTransaction().begin();
+      s.saveOrUpdate(singers);
+      s.getTransaction().commit();
+
+      s.getTransaction().begin();
+      Query query =
+          s.createQuery(
+              "update Singers set active=:active "
+                  + "where lastName=:lastName and firstName=:firstName");
+      query.setParameter("active", false);
+      query.setParameter("lastName", "Cord");
+      query.setParameter("firstName", "David");
+      query.executeUpdate();
+      s.getTransaction().commit();
+
+      System.out.println("Updated singer: " + s.get(Singers.class, singers.getId()));
+    }
+  }
+
+  public void testHqlList() {
+    try (Session s = hibernateConfiguration.openSession()) {
+      Query query = s.createQuery("from Singers");
+      List<Singers> list = query.list();
+      System.out.println("Singers list size: " + list.size());
+
+      query = s.createQuery("from Singers order by fullName");
+      query.setFirstResult(2);
+      list = query.list();
+      System.out.println("Singers list size with first result: " + list.size());
+
+      /* Current Limit is not supported. */
```

Comment:
Marked this as not supported as the change is not published in PROD. Will update this when the change is published

Reply:
Ack